src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/cluster/NodeKey.java

```diff
@@ -36,13 +38,25 @@ public String getHostAddress() {
   public boolean equals(Object obj) {
     if (obj instanceof NodeKey) {
       NodeKey key = (NodeKey)obj;
-      return nodeId.equals(key.getNodeId());
+      return nodeId.equals(key.getNodeId()) && hostAddress.equals(key.getHostAddress());
```

Comment:
Great that we updated the `equals` and `hashCode` method here to include `hostAddress`